### PR TITLE
Enable CUDA 12.0 support

### PR DESCRIPTION
**Pull request by JanuszL**
- adds CUDA 12.0 based build and enables it in tests
- bumps up the LLVM version used in DALI deps docker build image to 15.x
- bumps up (or just reverts to the default) the gcc version in the DALI deps image

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:
**New feature** (*non-breaking change which adds functionality*)

## Description:
- adds CUDA 12.0 based build and enables it in tests
- bumps up the LLVM version used in DALI deps docker build image to 15.x
- bumps up (or just reverts to the default) the gcc version in the DALI deps image

### Affected modules and functionalities:
- new DALI variant is available

### Key points relevant for the review:
- ?

### Tests:

- [x] Existing tests apply
  - all tests
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [x] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -165,7 +170,7 @@ CUDA 9 build is provided up to DALI 0.22.0.
 
 CUDA 10 build is provided up to DALI 1.3.0.
 
-CUDA 10.2 build is provided starting from DALI 1.4.0.
+CUDA 10.2 build is provided starting from DALI 1.4.0 up to DALI 1.20.
 
 CUDA 11 build is provided starting from DALI 0.22.0.
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/installation.rst]: Done

File: README.rst [hunk 2]
```diff
@@ -134,9 +134,9 @@ GitHub issue.
 Installing DALI
 ---------------
 
-To install the latest DALI release for the latest CUDA version (11.x)::
+To install the latest DALI release for the latest CUDA version (12.x)::
```
Comment (JanuszL) [inline README.rst:137]: Should we update this now?

**Review by mzient — APPROVED** [review]

**Review by szalpal — APPROVED** [review]

File: README.rst (hunk 2 above)
Comment (stiepan) [reply README.rst:137]: Good point, it may be miesleading for over 1.5 month.

Comment (JanuszL) [reply README.rst]: Reverted

**Review by stiepan — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Pull request merged**